armory/scenarios/poison.py

```diff
@@ -0,0 +1,364 @@
+        val_data = config_loading.load_dataset(
+            self.config["dataset"], split=self.validation_split, **self.dataset_kwargs
+        )
+        for x, y in tqdm(val_data, desc="Validation"):
+            # Ensure that input sample isn't overwritten by classifier
+            x.flags.writeable = False
+            y_pred = self.estimator.predict(x)
+            self.benign_validation_metric.add_results(y, y_pred)
+            y_pred_target_class = y_pred[y == self.source_class]
+            if len(y_pred_target_class):
+                self.target_class_benign_metric.add_results(
+                    [self.source_class] * len(y_pred_target_class), y_pred_target_class
+                )
+
+    def test(self):
+        if self.use_poison:
+            logger.info("Testing on poisoned test data")
+            test_data = config_loading.load_dataset(
+                self.config["dataset"], split=self.test_split, **self.dataset_kwargs
+            )
+            for x, y in tqdm(test_data, desc="Testing"):
+                x, _ = self.test_poisoner.poison(x, y, fraction=1.0)
+                # Ensure that input sample isn't overwritten by classifier
+                x.flags.writeable = False
+                y_pred = self.estimator.predict(x)
+                self.poisoned_test_metric.add_results(y, y_pred)
+
+                y_pred_targeted = y_pred[y == self.source_class]
+                if len(y_pred_targeted):
+                    self.poisoned_targeted_test_metric.add_results(
+                        [self.target_class] * len(y_pred_targeted), y_pred_targeted
+                    )
+
+    def finalize(self):
+        logger.info(
+            f"Unpoisoned validation accuracy: {self.benign_validation_metric.mean():.2%}"
+        )
+        logger.info(
+            f"Unpoisoned validation accuracy on targeted class: {self.target_class_benign_metric.mean():.2%}"
+        )
+        results = {
+            "benign_validation_accuracy": self.benign_validation_metric.mean(),
+            "benign_validation_accuracy_targeted_class": self.target_class_benign_metric.mean(),
+        }
+        if self.use_poison:
+            results["poisoned_test_accuracy"] = self.poisoned_test_metric.mean()
+            results[
+                "poisoned_targeted_misclassification_accuracy"
+            ] = self.poisoned_targeted_test_metric.mean()
+            logger.info(f"Test accuracy: {self.poisoned_test_metric.mean():.2%}")
+            logger.info(
+                f"Test targeted misclassification accuracy: {self.poisoned_targeted_test_metric.mean():.2%}"
+            )
+        self.results = results
+
+    def _evaluate(self) -> dict:
+        """
+        Evaluate a config file for classification robustness against attack.
+        """
+        self.set_random_seed()
+        self.load_model()
+        self.load_defense()
+        self.load_train_dataset()
+        self.poison_dataset()
```

Comment:
I think ```self.load_attack()``` needs to be called before this line -- the ```poison_dataset``` method references self.use_poison, which is initialized in ```self.load_attack()```